app/controllers/ReportingController.scala

```diff
@@ -120,6 +130,13 @@ trait ReportingController extends BaseController {
     }
   }
 
+  private def enrichReport(candidates: List[CandidateProgressReportItem], postcodes: Map[String, String]):
+  Future[List[CandidateProgressReportItem]] = {
+
+    Future(candidates.map(candidate => candidate.copy(
```

Comment:
My intention was to actually spin this in another thread and it might be quite demanding from a computational point of view. The postcode map where we do the lookup, will have basically all the users with a postcode so could be quite chubby. But if you think that might not be an issue, I'm happy to use Future.successful